optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectEditor.java

```diff
@@ -121,7 +129,20 @@ protected void loadDataObject( DataObject dataObject ) {
         this.comparatorObject = null;
         if ( dataObject != null ) {
             boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
-            boolean hasPlanningSolution = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_SOLUTION_ANNOTATION ) != null;
+            boolean hasPlanningSolution = dataObject.getAnnotation( PLANNING_SOLUTION_ANNOTATION ) != null;
+
+            if ( !hasPlanningSolution ) {
+                dataModelerService.call( new RemoteCallback<List<Path>>() {
+                    @Override
+                    public void callback( List<Path> paths ) {
+                        view.enablePlanningSolutionCheckBox( paths.isEmpty() );
```

Comment:
@mcimbora while testing in development we're blind to latency. There are ways to simulated latency locally. IIRC on linux level, but maybe also through a firefox/chrome extension (which is probably easier to use). I recommend testing with a latency of 300ms per request (regardless of the size of the request, so bandwidth stays unlimited). This will immediately expose n+1 query problems, as those will have a latency of n times 300ms, resulting in very long visible delays.

We need to do this kind of testing at least once in every Final release cycle (just like findbugs, running coverage, etc). 

@yurloc Maybe we can even automate it by adding latency in selenium tests? Not sure if this is possible.